examples/CAN_Example/CAN_Example.ino

```diff
@@ -0,0 +1,172 @@
+    uint8_t value[4];
+} CAN_writeRegisterResponseMessage;
+
+CAN_writeRegisterResponseMessage writeRegisterResponseMessage;
+
+void setup() {
+
+  while (!Serial) {
+    // Wait for a Serial connection...
+  }
+
+  Serial.begin(115200);
+
+  // Setup SPI controller (using Raspberry Pi Pico)
+  SPI1.setSCK(MCP2515_SCK);
+  SPI1.setTX(MCP2515_MOSI);
+  SPI1.setRX(MCP2515_MISO);
+  SPI1.setCS(MCP2515_CS);
+  SPI1.begin();
+
+  Serial.println("Configure ACAN2515");
+
+  ACAN2515Settings settings(QUARTZ_FREQUENCY, 100UL * 1000UL);  // CAN bit rate 100 kb/s
+  settings.mRequestedMode = ACAN2515Settings::NormalMode ; // Select Normal mode
+  const uint16_t errorCode = can.begin(settings, onCanRecieve );
+
+  if (errorCode == 0) {
+    Serial.print("Actual bit rate: ");
+    Serial.print(settings.actualBitRate());
+  } else {
+    Serial.print(errorCode);
+  }
+
+  // Create CONBus registers
+  // The addresses can be anything from 0 to 255
+  conbus.addRegister(0, &useOven);
+  conbus.addRegister(4, &numberOfOvenRacks);
+  conbus.addRegister(21, &ovenTemperatureSetpoint);
+
+  Serial.print("useOven: ")
+  Serial.println(useOven);
+
+  conbus.writeRegister(0, true);
+
+  Serial.print("useOven: ")
+  Serial.println(useOven);
+
+  Serial.print("ovenTemperatureSetpoint: ")
+  Serial.println(ovenTemperatureSetpoint);
+
+  conbus.writeRegister(21, 400.0f);
+
+  Serial.print("ovenTemperatureSetpoint: ")
+  Serial.println(ovenTemperatureSetpoint);
+}
+
+void loop() {
+  if (awaiting_read_response) {
+    awaiting_read_response = false;
+
+    readRegisterResponseMessage.registerAddress = register_to_fetch;
+    conbus.readRegisterBytes(register_to_fetch, readRegisterResponseMessage.value, readRegisterResponseMessage.length);
+
+    outFrame.id = 1100 + DEVICE_ID;
+    outFrame.len = sizeof(readRegisterResponseMessage);
+    memcpy(outFrame.data, &readRegisterResponseMessage, sizeof(readRegisterResponseMessage));
+
+    const bool ok = can.tryToSend(outFrame);
+  }
+
+  if (awaiting_write_response) {
+    awaiting_write_response = false;
+
+    outFrame.id = 1300 + DEVICE_ID;
+    outFrame.len = sizeof(writeRegisterResponseMessage);
+    memcpy(outFrame.data, &writeRegisterResponseMessage, sizeof(writeRegisterResponseMessage));
+
+    const bool ok = can.tryToSend(outFrame);
+  }
+}
+
+void onCanRecieve() {
+  can.isr();
+  can.receive(frame);  
+
+  // CONBus read register
+  if (frame.id == (1000 + DEVICE_ID)) {
+    awaiting_fetch = true;
+
+    readRegisterMessage = *(*CAN_readRegisterMessage)frame.data;
+    register_to_fetch = readRegisterMessage.registerAddress;
+  }
+
+  // CONBus write register
+  if (frame.id == (1200 + DEVICE_ID)) {
+
+    writeRegisterMessage = *(*CAN_writeRegisterMessage)frame.data;
+    conbus.writeRegisterBytes(writeRegisterMessage.registerAddress, writeRegisterMessage.value, writeRegisterMessage.length);
+  
+    memcpy(&writeRegisterResponseMessage, *writeRegisterMessage, sizeof(writeRegisterResponseMessage));
```

Comment:
Need to set awaiting_write_response to true

Reply:
Also, probably shouldn't be doing memcpy in interrupts like this even if it is only <= 8 bytes